Sources/NostrSDK/Events/Channels/CreateChannelEvent.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  CreateChannelEvent.swift
+//
+//
+//  Created by Konstantin Yurchenko, Jr on 9/11/24.
+//
+
+import Foundation
+
+/// Create a public chat channel.
+/// See [NIP-28](https://github.com/nostr-protocol/nips/blob/master/28.md#kind-40-create-channel).
+public class CreateChannelEvent: NostrEvent {
+    
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+    
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    required init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    required init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), pubkey: String) {
+        super.init(kind: kind, content: content, tags: tags, createdAt: createdAt, pubkey: pubkey)
+    }
+
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(id: String, pubkey: String, createdAt: Int64, kind: EventKind, tags: [Tag], content: String, signature: String?) {
+        super.init(id: id, pubkey: pubkey, createdAt: createdAt, kind: kind, tags: tags, content: content, signature: signature)
+    }
+
+    init(content: String, tags: [Tag], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
+        try super.init(kind: Self.kind, content: content, tags: tags, createdAt: createdAt, signedBy: keypair)
+    }
+    
+    class var kind: EventKind {
+        .channelCreation
+    }
+}
+
+public extension EventCreating {
+    func createChannelEvent(withContent content: String, signedBy keypair: Keypair) throws -> CreateChannelEvent {
```

Comment:
I'm in the process of deprecating the `EventCreating` protocol actually, and moving towards a `Builder` pattern. The former is too rigid for building Nostr events, whereas the latter gives us a lot of flexibility.

See https://github.com/nostr-sdk/nostr-sdk-ios/pull/175/files#diff-c87e105b65b1d7c9512acf2ee8c7e3fc7582648811025c959a37b36d97403b67R170 as an example.

Also, it would be better to have an API where the developer can specify the basic channel metadata (name, about, picture and relays) rather than needing to enter in raw JSON in `content`.
https://github.com/nostr-protocol/nips/blob/master/28.md#kind-40-create-channel

Reply:
Hi, @tyiu, thanks for your time and looking over my commit! 

Yes, I am on board with the later. The interface in this PR needs further definition and polish. 
I created this PR because I am using your SDK with my project (SwiftUI/Swift 5). Thanks for putting it together!
So far, I was able to implement and successfully field test Encrypted Message, Group Chat and Key Management features. 
As far as, my PR goes I will clean it up soon for the linter. I am thinking doing it in parallel while I’m working on replies for my chat application. 

I don't quite understand what you mean by ‘deprecating EventCreating protocol’. I do use it a lot but I did find it somewhat rigid.
I also think there needs to be some better way to manage subscriptions. I do have them on about 5 Views on my app. Yikes...! 🦟

https://github.com/SkatePay/skatepay/blob/main/SkateConnect/SkateConnect/Views/Skatepark/Lobby/SpotFeed/ChannelFeed.swift#L145

Regards!

Reply:
When I say deprecating the `EventCreating` protocol in favor of the `Builder` protocol, I mean it would look something like this when creating this event:

```swift
try CreateChannelEvent.Builder()
    .channelMetadata(channelMetadata, merging: rawChannelMetadata)
    .build(signedBy: keypair)
```

Take a look at what we do here for kind 0, as an example:
https://github.com/nostr-sdk/nostr-sdk-ios/blob/8b40eb98a4268255b8980bddddf40125b0d32637/Sources/NostrSDK/Events/MetadataEvent.swift#L161-L192

Reply:
@masterial Turns out the SwiftLint errors were on the main branch, due to how CI pulls in the latest SwiftLint version and it had some breaking changes in what it validates. I just fixed the issue on the main branch so the lint errors should go away after you pull it into your branch.

Reply:
Merged, thanks for the heads up, @tyiu!
I plan to address other issues you've highlighted very soon

Reply:
Hi Terry, @tyiu 

I hope you're doing well. I'm reaching out because I'm delving into the world of Nostr SDK implementation, and I've encountered a few hurdles that I believe you might help me navigate:

Relay Connections: I'm experiencing issues with maintaining persistent relay connections. There seems to be a timeout problem. Could you guide me on how to best handle this? Specifically, what's the recommended approach for reconnecting relays when they drop?
Relay Pool Lifecycle: I'm unsure about managing the lifecycle of the relay pool. What are the best practices here? Should I be constantly monitoring and refreshing connections, or is there a more efficient way?
Debugging Events: Currently, there's a lot of event debugging happening in my setup which I'd like to minimize. Is there a way to disable or manage this more efficiently?

On another note, I'm excited to share that my app, [SkateConnect](https://support.skatepark.chat/) (also known as SkatePay), has just been released on the App Store. We're now focusing on enhancing its robustness, particularly with connection persistence. Given your expertise, I'd love to discuss potential collaborations or advisory roles concerning Nostr integration. If you're open to it, I'd appreciate discussing this further, perhaps exploring how we might make Nostr work more seamlessly within SkateConnect. (We can talk over nostr too, not here).

Looking forward to your thoughts and any advice you can offer!

Best regards!

PS. Still got some work to do before I polish this PR.